superset/databases/api.py

```diff
@@ -838,6 +848,70 @@ def validate_sql(self, pk: int) -> FlaskResponse:
         except DatabaseNotFoundError:
             return self.response_404()
 
+    @expose("/oauth2/", methods=["GET"])
+    @statsd_metrics
+    @event_logger.log_this_with_context(
+        action=lambda self, *args, **kwargs: f"{self.__class__.__name__}.oauth",
+        log_to_statsd=False,
+    )
+    def oauth2(self) -> FlaskResponse:
+        """
+        ---
+        get:
+          summary: >-
+            Receive user-level authentication tokens from OAuth
+          description: ->
+            Receive and store user-level authentication tokens from OAuth
+          parameters:
+          - in: query
+            name: state
+          - in: query
+            name: code
+          - in: query
+            name: scope
+          - in: query
+            name: error
+          responses:
+            200:
+              description: A dummy self-closing HTML page
+              content:
+                text/html:
+                  schema:
+                    type: string
+            400:
+              $ref: '#/components/responses/400'
+            500:
+              $ref: '#/components/responses/500'
+        """
+        parameters = request.args.to_dict()
+        if "error" in parameters:
+            raise Exception(parameters["error"])  # XXX: raise custom SIP-40 exception
```

Comment:
is this a todo?

Reply:
Yes! `XXX` is a traditional way of adding a todo (vim will highlight it by default, eg).

https://stackoverflow.com/a/5456703/807118

Reply:
Normally I use it for temporary todos, while still working on the PR.